storage/client/src/main/java/com/azure/storage/queue/models/QueueProperties.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.storage.queue.models;
+
+import java.util.Map;
+
+public final class QueueProperties {
+    /**
+     * The metadata property.
+     */
+    private final Map<String, String> metadata;
+
+    /**
+     * The approximate number of messages in the queue. This number is not
+     * lower than the actual number of messages in the queue, but could be
+     * higher.
+     */
+    private final int approximateMessagesCount;
+
+    public QueueProperties(Map<String, String> metadata, int approximateMessagesCount) {
+        this.metadata = metadata;
+        this.approximateMessagesCount = approximateMessagesCount;
+    }
+
+    public Map<String, String> metadata() {
+        return this.metadata; // Clone this?
+    }
+
+    public int approximateMessagesCount() {
+        return approximateMessagesCount;
+    }
```

Comment:
I'm on the fence about whether this class should be fluent or POJO, but I'm leaning towards it being ok to be fluent. Thoughts?

Reply:
Given that class shouldn't be mutated, since there is no good use case for it to be, I'd learn towards an immutable POJO.

Will update getter names.